src/schedule/timer_domain.c

```diff
@@ -119,6 +119,7 @@ static int timer_domain_register(struct ll_schedule_domain *domain,
 #ifdef __ZEPHYR__
 	void *stack;
 	char *qname;
+	struct k_thread *thread;
```

Comment:
I'm assuming that 
1) all threads are idle when there is no work i.e. we are in WFI
2) EDF work is preemptable by LL pipeline tasks ?

I also want to confirm that

3) All IPC is premptable (including via IDC)
4) Pipeline work should mostly be non pre-emptable, but we should also support pre-emtable pipelines and components. i.e. KWD media decode.
5) trace work will be pre-emptable.

Reply:
1. (idle) - should be, yes
2. I think so. Under Zephyr both are implemented as tasks, but the timer domain (for LL scheduling) uses `-CONFIG_NUM_COOP_PRIORITIES` (e.g. -16) as priority, whereas EDF uses 1. Natively with XTOS the LL timer domain handler is run in the timer interrupt context, so, with Zephyr the highest available priority is used.
3. "real" IPC (host->DSP) is unchanged, seems to be mapped to Zephyr task priority 0. IDC runs in P4WQ threads and those use a fixed priority, which is the same as for LL scheduling above - the highest priority of `-CONFIG_NUM_COOP_PRIORITIES`. I guess we have to discuss this...
4. Timer domain pipelines are threads with the highest priority ATM, I think the DMA domain LL tasks are so far unchanged, i.e. called in the DMA interrupt context, but they probably have never been tested with Zephyr, so might be broken.
5. Not sure about Zephyr printk / logging. I think they use some threaded context, e.g. for logging I see a thread created with the `K_LOWEST_APPLICATION_THREAD_PRIO` priority, which is... rather low (depends on Kconfig). But `printk()` seems to run synchronously.

In general, this patch set isn't touching much of the above, it only does one thing WRT threading / priorities: implementing IDC via P4WQ, and there the present API doesn't allow us to select priority. If we need that, we have to discuss that with Zephyr.